### DISCUSSION TITLE
Endpoints without param get not queried with only explicit phase

### DISCUSSION THREAD

**Discussion by TradingMarius** [schemathesis/schemathesis#1356] [Q&A] [answered]
Hi,
I want to use Schemathesis to test our api without using randomly generated parameters. For this, I am using `--hypothesis-phases=explicit` and it works for all endpoints where there is a parameter defined. But when I have an endpoint without a parameter e.g. `GET /all` there are no tests executed. Is there anyway to mitigate that?
Thanks in advance!!

**Comment by Stranger6667** [comment, marked as answer]
Hi @TradingMarius 

Thanks for asking :)

I think that #1323 is closely related and I was thinking to add explicit "SKIPPED" indicators for such cases. Though, it might be indeed better to generate an "empty" test case (as there are no parameters). To make it work we need to ensure that [this list](https://github.com/schemathesis/schemathesis/blob/master/src/schemathesis/specs/openapi/examples.py#L154) contains at least one strategy if there are no examples. Something like this:

```python
def get_strategies_from_examples(operation: APIOperation, examples_field: str = "examples") -> List[SearchStrategy[Case]]:
    ...
    if not strategies:
        strategies.append(get_case_strategy(operation=operation))
    return strategies
```

I'll take a look at this today - if there are no corner cases, then I think this change could be released in the next version

**Reply by Stranger6667** [reply]
Currently, it could be solved via the [before_add_examples](https://schemathesis.readthedocs.io/en/stable/extending.html#before-add-examples) hook:

```python
import schemathesis
from schemathesis import Case
from typing import List


def before_add_examples(
    context: schemathesis.hooks.HookContext,
    examples: List[Case],
) -> None:
    if not examples:
        examples.append(Case(operation=context.operation))
```

P.S. The link above also contains instructions on how to attach hooks

**Reply by TradingMarius** [reply]
Thanks for the quick reply!
I am fine with either way.
I also agree that generating an empty test case seems to be the better choice (for me at least ^^)

**Reply by TradingMarius** [reply]
Thanks again!
I will try this out.

**Reply by Stranger6667** [reply]
You're very welcome :)
